### PR TITLE
Added verification towards disk space

### PR DESCRIPTION
**Pull request by chria**

### CODE REVIEW

File: src/common/io.py [hunk 1]
```diff
@@ -42,6 +43,15 @@
 
 SYS_LITTLE_ENDIAN = sys.byteorder == 'little'
 NCP_LARGE = 5000
+DISK_PROTECTION = 50*1024**2 #50mb
+
+#Get the available disk space at a given local path (minus 50mb to make sure we do not run out of space)
+def get_available_disk_space(local_path):
```
Comment (PeterMeisrimelModelon) [inline src/common/io.py:49]:
```suggestion
def get_available_disk_space(local_path):
    """Get the available disk space at a given local path (minus 50mb to make sure we do not run out of space)."""
```
Might as well make this a docstring.

File: src/common/io.py [hunk 2]
```diff
@@ -42,6 +43,15 @@
...
+    try:
+        free_space = disk_usage(local_path).free - DISK_PROTECTION
```
Comment (PeterMeisrimelModelon) [inline src/common/io.py:51]: Not sure I understand what the rational behind `DISK_PROTECTION` is. 
Can we add some comment here?

File: src/common/io.py [hunk 3]
```diff
@@ -42,6 +43,15 @@
...
+    except (FileNotFoundError, TypeError):
```
Comment (PeterMeisrimelModelon) [inline src/common/io.py:52]: Can we add some comments here on what these are meant to catch? This is not evident to me, even after checking the documentation of `disk_usage`.

**Review by PeterMeisrimelModelon — CHANGES_REQUESTED** [review]

File: src/common/io.py (hunk 3 above)
Comment (chria) [reply src/common/io.py:60]: Added doc.

File: src/common/io.py (hunk 2 above)
Comment (chria) [reply src/common/io.py:57]: Added doc.

**Review by PeterMeisrimelModelon — APPROVED** [review]
Thanks for adding documentation! 

Can you also add a changelog entry? Good to merge then!

**Pull request merged**